apidoc/Titanium/UI/WebView.yml

```diff
@@ -368,6 +463,69 @@ events:
       - name: url
         summary: The URL of the web document that is stopped.
 
+  - name: message
+    summary: Fired when a script message is received from a webpage.
+    description: |
+        This event get fired when you have added message handler using method 'addScriptMessageHandler'
+        and  webpage sends message to it.
+    platforms: [ipad, iphone]
+    since: "7.4.0"
+    properties:
+      - name: url
+        summary: URL of the web document being loaded.
+        type: String
+      - name: body
+        summary: The body of message sent from webview.
+        type: String
+      - name: name
+        summary: The name of the message handler to which the message is sent.
+        type: String
+      - name: isMainFrame
+        summary: |
+            A Boolean value indicating whether the frame is the web site's main frame or a subframe.
+        type: Boolean
+
+  - name: progress
+    summary: Fired when webpage download progresses.
+    description: |
+        This value ranges from 0.0 to 1.0 based on the total number of bytes expected to be received, 
+        including the main document and all of its potential subresources.
+    platforms: [ipad, iphone]
+    since: "7.4.0"
+    properties:
+      - name: value
+        summary: An estimate of what fraction of the current navigation has been loaded.
+        type: Number
+      - name: url
+        summary: URL of the web document being loaded.
+        type: String
+
+  - name: redirect
+    summary: Fired when a web view receives a server redirect.
+    platforms: [ipad, iphone]
+    since: "7.4.0"
+    properties:
+      - name: title
+        summary: Page title of webpage.
+        type: String
+      - name: url
+        summary: URL of the web document being loaded.
+        type: String
+
+  - name: handleurl
+    summary: Fired when a web view receives a server redirect.
+    description: |
+            See the example section "Usage of allowedURLSchemes and handleurl  in iOS".
+    platforms: [ipad, iphone]
+    since: "7.4.0"
+    properties:
+      - name: handler
+        summary: Page title of webpage.
+        type: String
+      - name: url
+        summary: URL of the web document being loaded.
+        type: String
+
```

Comment:
I think the `handleurl` summary is wrong. This even is fired for every URL and before the WebView does the HTTP request, right?

Also, doesn't `handleurl` require immediate feedback? So, if "tiapp.xml" property `run-on-main-thread` is set to `false`, then there is no way to respond to this in-time. We'll run into the same issue as our `onLink` feature in PR #9459.

Feedback?

Reply:
the handler is a completion handler that can be called async, passing a constant to allow or disallow the request. there is an example in the ti.wlwebview docs